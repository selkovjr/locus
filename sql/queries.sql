--
--  Locus datatype test
--
-- Test operators on the table
SELECT count(*) FROM test_locus WHERE p && '21';
SELECT count(*) FROM test_locus WHERE p && 'chr21:10600000-12608058';
SELECT count(*) FROM test_locus WHERE p <& 'chr21:28800000-30000000';
SELECT count(*) FROM test_locus WHERE p << 'chr21:28800000-30000000';
SELECT count(*) FROM test_locus WHERE p >> 'chr21:28800000-30000000';
SELECT count(*) FROM test_locus WHERE p &> 'chr21:28800000-30000000';
SELECT count(*) FROM test_locus WHERE p <@ 'chr21:28800000-30000000';
SELECT count(*) FROM test_locus WHERE p <@ '<all>:28800000-30000000';

-- Test sorting
SELECT * FROM test_locus WHERE range(p) && '[29475000, 29475000]' ORDER by p;

-- Test lower, upper, and center functions
SELECT DISTINCT p, lower(p), center(p), upper(p) FROM test_locus WHERE length(p) > 10000 AND contig(p) IN ('13', '21', 'X') ORDER BY p;
