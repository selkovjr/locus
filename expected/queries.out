--
--  Locus datatype test
--
-- Test operators on the table
SELECT count(*) FROM test_locus WHERE p && '21';
 count
-------
 14161
(1 row)

SELECT count(*) FROM test_locus WHERE p && 'chr21:10600000-12608058';
 count
-------
   208
(1 row)

SELECT count(*) FROM test_locus WHERE p <& 'chr21:28800000-30000000';
 count
--------
 270982
(1 row)

SELECT count(*) FROM test_locus WHERE p << 'chr21:28800000-30000000';
 count
--------
 261810
(1 row)

SELECT count(*) FROM test_locus WHERE p >> 'chr21:28800000-30000000';
 count
-------
 62578
(1 row)

SELECT count(*) FROM test_locus WHERE p &> 'chr21:28800000-30000000';
 count
-------
 63971
(1 row)

SELECT count(*) FROM test_locus WHERE p <@ 'chr21:28800000-30000000';
 count
-------
    68
(1 row)

SELECT count(*) FROM test_locus WHERE p <@ '<all>:28800000-30000000';
 count
-------
 10520
(1 row)

-- Test sorting
SELECT * FROM test_locus WHERE range(p) && '[29475000, 29475000]' ORDER by p;
            p
-------------------------
 chr1:29474255-29475739
 chr1:29474365-29475739
 chr6:29474946-29475446
 chr16:29474666-29475277
 chr16:29474666-29475662
 chr21:29474869-29475089
 chr21:29474869-29475110
 chr21:29474869-29475283
 chr21:29474869-29475560
 chr21:29474869-29475900
(10 rows)

-- Test lower, upper, and center functions
SELECT DISTINCT p, lower(p), center(p), upper(p) FROM test_locus WHERE length(p) > 10000 AND contig(p) IN ('13', '21', 'X') ORDER BY p;
             p             |   lower   |  center   |   upper   
---------------------------+-----------+-----------+-----------
 chr13:23902965-23915830   |  23902965 |  23909397 |  23915830
 chr13:42803235-42817033   |  42803235 |  42810134 |  42817033
 chr13:52010562-52025311   |  52010562 |  52017936 |  52025311
 chr13:53261869-53275045   |  53261869 |  53268457 |  53275045
 chr13:67775147-67802708   |  67775147 |  67788927 |  67802708
 chr13:88327636-88348528   |  88327636 |  88338082 |  88348528
 chr13:99047481-99059749   |  99047481 |  99053615 |  99059749
 chr13:102363154-102375318 | 102363154 | 102369236 | 102375318
 chr13:103381719-103402739 | 103381719 | 103392229 | 103402739
 chr21:35260456-35272166   |  35260456 |  35266311 |  35272166
 chr21:35467162-35478562   |  35467162 |  35472862 |  35478562
 chr21:38884214-38896294   |  38884214 |  38890254 |  38896294
 chr21:38979678-38997787   |  38979678 |  38988732 |  38997787
 chr21:41032487-41045065   |  41032487 |  41038776 |  41045065
 chr21:42547158-42557167   |  42547158 |  42552162 |  42557167
 chrX:18646491-18656528    |  18646491 |  18651509 |  18656528
 chrX:23410648-23422490    |  23410648 |  23416569 |  23422490
 chrX:71363103-71373137    |  71363103 |  71368120 |  71373137
 chrX:73012040-73049067    |  73012040 |  73030553 |  73049067
 chrX:73061217-73071428    |  73061217 |  73066322 |  73071428
 chrX:73061217-73072559    |  73061217 |  73066888 |  73072559
 chrX:131503345-131513706  | 131503345 | 131508525 | 131513706
 chrX:137696892-137715148  | 137696892 | 137706020 | 137715148
 chrX:139856568-139875381  | 139856568 | 139865974 | 139875381
(24 rows)

